lib/Interpreter/CIFactory.cpp

```diff
@@ -316,14 +316,9 @@ namespace {
       }
 
     #if defined(__GLIBCXX__)
-      // Avoid '__float128 is not supported on this target' errors
```

Comment:
What does this comment add? We'd *always* pass `-std=c++11`, right? I find this comment very confusing - it's almost like saying "Avoid '__float128 is not supported on this target' errors: pass `-x c++`". I understand clang might have had at some point this issue, but I don't think there is a close enough causal connection to put it here.

On the contrary: the fact that cling on GCC6 / mac complains
```
/usr/local/Cellar/gcc@6/6.3.0/lib/gcc/6/gcc/x86_64-apple-darwin15.6.0/6.3.0/../../../../../../include/c++/6.3.0/type_traits:344:39: error: __float128 is not supported on this target
```
means likely that we are missing an arch flag - completely unrelated to the standard version!

I.e. I acknowledge that it's broken now; I'll try to figure out why. This here is not the *real* reason.